src/python/grpcio_tests/tests/unit/_xds_credentials_test.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2020 The gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests xDS server and channel credentials."""
+
+import unittest
+
+import logging
+from concurrent import futures
+
+import grpc
+import grpc.experimental
+from tests.unit import test_common
+from tests.unit import resources
+
+
+class _GenericHandler(grpc.GenericRpcHandler):
+
+    def service(self, handler_call_details):
+        return grpc.unary_unary_rpc_method_handler(
+            lambda request, unused_context: request)
+
+
+class XdsCredentialsTest(unittest.TestCase):
+
+    def test_xds_creds_fallback_ssl(self):
+        # Since there is no xDS server, the fallback credentials will be used.
+        # In this case, SSL credentials.
+        server = grpc.server(futures.ThreadPoolExecutor())
+        server.add_generic_rpc_handlers((_GenericHandler(),))
+        server_fallback_creds = grpc.ssl_server_credentials(
+            ((resources.private_key(), resources.certificate_chain()),))
+        server_creds = grpc.xds_server_credentials(server_fallback_creds)
+        port = server.add_secure_port("localhost:0", server_creds)
+        server.start()
```

Comment:
optional: the difference between created servers is `xds` and the given server credentials, we could create a factory method or a context manager for the server. It might help people reading these test cases to spot what is the focus.